From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: bchew <basil.chew@intel.com>
Date: Wed, 22 Mar 2023 11:32:17 +0000
Subject: [PATCH] Switch to using split OVMF

Changes:
- updated setup script to copy split OVMF files
- configured ini script to use split OVMF
---
 scripts/setup_civ_ini.sh | 3 ++-
 scripts/setup_host.sh    | 9 ++++++++-
 2 files changed, 10 insertions(+), 2 deletions(-)

diff --git a/scripts/setup_civ_ini.sh b/scripts/setup_civ_ini.sh
index aab7697..0155cca 100755
--- a/scripts/setup_civ_ini.sh
+++ b/scripts/setup_civ_ini.sh
@@ -13,7 +13,8 @@ function setup_civ_ini() {
         filename=$(ls caas-flashfiles*)
         sed -i "/^\[global\]$/,/^\[/ s#^name.*#name=civ-sriov#" $USER_HOME/.intel/.civ/civ-sriov.ini
         sed -i "/^\[global\]$/,/^\[/ s#^flashfiles.*#flashfiles=$CIV_WORK_DIR\/$filename#" $USER_HOME/.intel/.civ/civ-sriov.ini
-        sed -i "/^\[firmware\]$/,/^\[/ s#^path.*#path=$CIV_WORK_DIR\/OVMF.fd#" $USER_HOME/.intel/.civ/civ-sriov.ini
+        sed -i "/^\[firmware\]$/,/^\[/ s#^type.*#type=splited#" $USER_HOME/.intel/.civ/civ-sriov.ini
+        sed -i "/^\[firmware\]$/,/^\[/ s#^path.*#code=$CIV_WORK_DIR\/OVMF_CODE_android.fd\nvars=$CIV_WORK_DIR\/OVMF_VARS_android.fd#" $USER_HOME/.intel/.civ/civ-sriov.ini
         sed -i "/^\[disk\]$/,/^\[/ s#^path.*#path=$CIV_WORK_DIR\/android.qcow2#" $USER_HOME/.intel/.civ/civ-sriov.ini
         sed -i "/^\[graphics\]$/,/^\[/ s#^type.*#type=SRIOV#" $USER_HOME/.intel/.civ/civ-sriov.ini
 
diff --git a/scripts/setup_host.sh b/scripts/setup_host.sh
index cc697a8..1763f3c 100755
--- a/scripts/setup_host.sh
+++ b/scripts/setup_host.sh
@@ -142,7 +142,14 @@ function ubu_build_ovmf_gvt(){
     source ./edksetup.sh
     make -C BaseTools/
     build -b DEBUG -t GCC5 -a X64 -p OvmfPkg/OvmfPkgX64.dsc -D NETWORK_IP4_ENABLE -D NETWORK_ENABLE  -D SECURE_BOOT_ENABLE -D TPM_ENABLE
-    cp Build/OvmfX64/DEBUG_GCC5/FV/OVMF.fd ../OVMF.fd
+    if [ ! -d $CIV_WORK_DIR/ovmf ]; then
+        mkdir $CIV_WORK_DIR/ovmf
+    fi
+    cp Build/OvmfX64/DEBUG_GCC5/FV/OVMF_CODE.fd $CIV_WORK_DIR/ovmf/OVMF_CODE_civ.fd
+    cp Build/OvmfX64/DEBUG_GCC5/FV/OVMF_VARS.fd $CIV_WORK_DIR/ovmf/OVMF_VARS_civ.fd
+    ln -sf $CIV_WORK_DIR/ovmf/OVMF_CODE_civ.fd $CIV_WORK_DIR/OVMF_CODE_android.fd
+    cp $CIV_WORK_DIR/ovmf/OVMF_VARS_civ.fd $CIV_WORK_DIR/OVMF_VARS_android.fd
+
 
     if [ -d $CIV_GOP_DIR ]; then
         local gpu_device_id=$(cat /sys/bus/pci/devices/0000:00:02.0/device)
-- 
2.17.1

